src/dtx/dtx_common.c

```diff
@@ -905,23 +912,38 @@ dtx_handle_init(struct dtx_id *dti, daos_handle_t coh, struct dtx_epoch *epoch,
 
 	dth->dth_dkey_hash = 0;
 
-	if (daos_is_zero_dti(dti))
-		return 0;
+	if (!local) {
+		if (daos_is_zero_dti(dti))
+			return 0;
 
-	if (!dtx_epoch_chosen(epoch)) {
-		D_ERROR("initializing DTX "DF_DTI" with invalid epoch: value="
-			DF_U64" first="DF_U64" flags=%x\n",
-			DP_DTI(dti), epoch->oe_value, epoch->oe_first,
-			epoch->oe_flags);
-		return -DER_INVAL;
+		if (!dtx_epoch_chosen(epoch)) {
+			D_ERROR("initializing DTX " DF_DTI " with invalid epoch: value=" DF_U64
+				" first=" DF_U64 " flags=%x\n",
+				DP_DTI(dti), epoch->oe_value, epoch->oe_first, epoch->oe_flags);
+			return -DER_INVAL;
+		}
+		dth->dth_epoch       = epoch->oe_value;
+		dth->dth_epoch_bound = dtx_epoch_bound(epoch);
 	}
-	dth->dth_epoch = epoch->oe_value;
-	dth->dth_epoch_bound = dtx_epoch_bound(epoch);
 
-	if (dth->dth_modification_cnt == 0)
-		return 0;
+	rc = vos_dtx_rsrvd_init(dth);
+	if (rc != 0) {
+		D_ERROR("Failed to allocate space for scm reservations: rc=" DF_RC "\n", DP_RC(rc));
+		return rc;
+	}
 
-	return vos_dtx_rsrvd_init(dth);
+	if (local) {
```

Comment:
In this case, `dtx_handle_init()` won't produce initialized DTX handles in all the cases. In the case of the `DTX_LOCAL` it will be half-baked which may be confusing. I think it is better to keep it where it is so the role of the `dtx_handle_init()` function will remain unchanged.